### PR TITLE
Dg update

### PR DESCRIPTION
**Pull request by Thing1Thing2**

### CODE REVIEW

**Comment by sikai00** [comment]
hey, is it possible to resolve the conflict first, then ill help to take a look at the changes, thanks!

**Comment by ThomasHoooo** [comment]
May i check why are there so many commits? This shouldn't be the case.

**Comment by Thing1Thing2** [comment]
Changes were made after resolving the first conflict, so resolved again.

**Comment by ThomasHoooo** [comment]
> Changes were made after resolving the first conflict, so resolved again.

If you resolved correctly, this should only have one commit stating that you resolve conflicts. From my understanding, now your pr is trying to merge in all 24 commits which is odd.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -399,80 +450,45 @@ We chose to implement the changing of view panels through `CommandResult` due to
...
-
-<img src="images/AlternativeEditActivityDiagram.png" width="250" />
-
-#### Rationale
-LogicManager is responsible for coordinating both Model and Storage updates because Model and Storage should be kept as separate entities.
-
-[Proposed] Multiple possible prefixes per command feature
-
-In this proposed feature, the user is provided with  multiple possible prefixes for defining fields in a command.
+Purpose: In this proposed feature, the user is provided with  multiple possible prefixes for defining fields in a command.
```
Comment (ThomasHoooo) [inline docs/DeveloperGuide.md:457]: Can you help me understand this sentence better?

Comment (Thing1Thing2) [reply docs/DeveloperGuide.md:457]: Can use alternative prefixes such as `d/` and `b/` for defining the same field, say the birthday field of a client. Any suggestion on how to word better?

Comment (ThomasHoooo) [reply docs/DeveloperGuide.md:457]: Sorry I still do not understand this proposed feature. And does it really add value to our product? If not, I think it is okay to delete this whole portion

Comment (ThomasHoooo) [reply docs/DeveloperGuide.md:457]: Maybe the other team members can understand and give better suggestions

Comment (Thing1Thing2) [reply docs/DeveloperGuide.md:457]: I added this section so  I could try mentioning a proposed feature, as is allowed in dg.  Not sure how to proceed either.... don't mind if PR is closed.

Comment (sikai00) [reply docs/DeveloperGuide.md:457]: Do you mean like allowing users to customize prefixes?

Comment (Thing1Thing2) [reply docs/DeveloperGuide.md:457]: we provide alternative prefixes, but an extension to this feature can be to allow users to customize their own prefixes.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -212,6 +212,22 @@ Below is an activity diagram that summarises the execution of `viewClient`.
 
 ![ViewClientActivityDiagram](images/ViewClientActivityDiagram.png)
 
+#### 3.1.3 Edit Client feature
+
+Syntax: `editClient i/INDEX [n/NAME] [p/PHONE_NUMBER] [a/ADDRESS] [e/EMAIL] [b/BIRTHDAY] [pd/PRODUCT]`
```
Comment (sikai00) [inline docs/DeveloperGuide.md:217]: pd ....

**Pull request merged**